### PR TITLE
HTBHF-2125 Move smart-stub over to use Circle CI.

### PR DESCRIPTION
**Pull request by mikeSmith08**
This is quite a cut-down version of the build compared to the other Java services as it doesn't deploy to Prod or do Code coverage.

Also needed to fix the move to use the common config sub module for Checkstyle and PMD configuration files.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:11.0.4-jdk-stretch
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
```
Comment (YetAnotherMatt) [inline .circleci/config.yml]: We can move to caching ./gradle - i.e. a single cache. See htbhf-dwp-api

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            export APP_VERSION=`cat version.properties | grep "version" | cut -d'=' -f2`
+            echo "export APP_VERSION=${APP_VERSION}" >> $BASH_ENV
+            echo "export APP_PATH="build/libs/$APP_NAME-$APP_VERSION.jar"" >> $BASH_ENV
+            source $BASH_ENV
+            echo "APP_VERSION=${APP_VERSION}, APP_PATH=${APP_PATH}"
+      - run:
+          name: deploy to dev and staging
+          command: ./ci_scripts/ci_deploy.sh
+      - run:
+          name: set TRAVIS_BRANCH as this is required by shipkit
```
Comment (YetAnotherMatt) [inline .circleci/config.yml]: We can remove this as we're not publishing to bintray

File: .circleci/config.yml (hunk 1 above)
Comment (mikeSmith08) [reply .circleci/config.yml]: Done.

File: .circleci/config.yml (hunk 2 above)
Comment (mikeSmith08) [reply .circleci/config.yml]: Ok, have removed.

**Pull request merged**